src/layout/PlotXY/components/axis/axis.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { axisBottom, axisLeft, AxisScale } from 'd3-axis';
+import { select, selectAll } from 'd3-selection';
+import React, { useEffect } from 'react';
+
+export default function Axis(props: {
+  scale: AxisScale<number>;
+  orient: string;
+  ticks?: number;
+  transform: string;
+}) {
+  const { scale, orient, ticks, transform } = props;
+  const ref = React.createRef<SVGGElement>();
+  useEffect(() => {
+    const renderAxis = () => {
+      const node: any = ref.current;
+      let axis: any;
+
+      if (orient === 'bottom') {
+        axis = axisBottom(scale);
+      }
+      if (orient === 'left') {
+        axis = axisLeft(scale).ticks(ticks);
+      }
+      select(node).call(axis);
+    };
+    const updateAxis = () => {
+      if (orient === 'left') {
+        const axis: any = axisLeft(scale).ticks(ticks);
+        selectAll(`.${orient}`).call(axis);
+      }
+    };
+    renderAxis();
+    return () => {
+      updateAxis();
+    };
+  }, [orient, ref, scale, ticks]);
+
+  return <g ref={ref} transform={transform} className={`${orient} axis`} />;
+}
```

Comment:
@lpatiny can i use these axis or i have to change it?

Reply:
Please get rid of the `d3-selection` dependency and do not render anything in a `useEffect`

Reply:
can i use react-d3-utils package?

Reply:
Yes it's even better if you can use it

Since this package is not very well documented and not very mature, please open issues on the corresponding repo if you have questions or if you see opportunities for improvement

Reply:
I checked the package and had a problem while using it (issue added)
Do I have to create a new axis (from scratch or can I use other external packages)?

Reply:
@stropitek I created new axis from scratch ( I can add more props for styling axis but I kept it simple)

Reply:
Regarding the tick and label position, you can create a PR to fix it in react-d3-utils

Reply:
Okey no problem

Reply:
- PR added in react-d3-utils 
- Create axis with react-d3-utils  done